tests/test_primary.py

```diff
@@ -122,7 +122,7 @@ def create_primary_pinning_file():
   pinnings = json.load(
   open(TEST_PINNING_TEMPLATE_FNAME, 'r', encoding='utf-8')) 
 
-  fname_to_create = os.path.join(TEST_DATA_DIR, "pinned.json")
+  fname_to_create = TEST_TEMP_PINNING_FNAME
```

Comment:
This structure is a little convoluted.

- test_data/pinned.json already existed and was used by other modules as well as this one, previously
- test_data/pinned_template.json you're adding

You're reading in pinned_template.json, modifying it, and overwriting pinned.json. Then pinned.json stays in its modified form afterwards. Other test modules then load this modified form, but the modifications don't affect them because they, too, modify the same portions of the file. That's why this works, but there was no need to overwrite pinned.json in the first place. Better to use pinned.json as it was, just as test_secondary does, and modify that in memory to update it to point to the right temp directory. So I'll probably exclude this part in the squashed commit.